File: src/cs2/noflash.rs
use crate::config::Config;

use super::{player::Player, CS2};

impl CS2 {
    pub fn no_flash(&self, config: &Config) {
        let process = match &self.process {
            Some(process) => process,
            None => return,
        };

        if !config.misc.no_flash {
            return;
        }

        let local_player = match Player::local_player(process, &self.offsets) {
            Some(player) => player,
            None => return,
        };

        local_player.no_flash(process, &self.offsets, config.misc.max_flash_alpha);
    }
}

File: src/cs2/no_flash.cpp
#include "cs2/cs2.hpp"
#include "cs2/features.hpp"

void NoFlash() {
    const auto local_player_opt = Player::LocalPlayer();
    if (!local_player_opt.has_value()) {
        return;
    }
    Player local_player = local_player_opt.value();

    if (config.misc.no_flash) {
        local_player.NoFlash(config.misc.max_flash_alpha);
    } else {
        local_player.NoFlash(255.0f);
    }
}
